extension/src/heartbeat_agg.rs

```diff
@@ -80,6 +82,22 @@ impl HeartbeatTransState {
         }
     }
 
+    // In general we shouldn't need to change these creation time parameters, but if
+    // we're combining with another interval this may be necessary.
+    fn extend_covered_interval(&mut self, new_start: i64, new_end: i64) {
+        assert!(new_start <= self.start && new_end >= self.end);
```

Comment:
Can a user trigger this, and if so, don't we need an error message that references the user interface?  Should we have two assertions so it's immediately clear which error the user hit?

If it should be impossible, a comment like the one on line 93 would help.